deploy/osd-machine-api/OWNERS

```diff
@@ -0,0 +1,4 @@
+reviewers:
+- yithian
+approvers:
+- yithian
```

Comment:
We aren't doing sublevel approvers, so you can remove this. Also, please add me and @luis-falcon to this list